pkg/lib/database/database.go

```diff
@@ -37,13 +37,57 @@ type Query struct {
 }
 
 func injectDefaultFilters(filters []*model.BlockContentDataviewFilter) []*model.BlockContentDataviewFilter {
+	hasArchivedFilter, hasDeletedFilter, hasTypeFilter := hasDefaultFilters(filters)
+	if len(filters) > 0 && len(filters[0].NestedFilters) > 0 {
+		return addDefaultFiltersToNested(filters, hasArchivedFilter, hasDeletedFilter, hasTypeFilter)
+	}
+	return addDefaultFilters(filters, hasArchivedFilter, hasDeletedFilter, hasTypeFilter)
+}
+
+func addDefaultFiltersToNested(filters []*model.BlockContentDataviewFilter, hasArchivedFilter, hasDeletedFilter, hasTypeFilter bool) []*model.BlockContentDataviewFilter {
+	if filters[0].Operator == model.BlockContentDataviewFilter_And {
+		filters[0].NestedFilters = addDefaultFilters(filters[0].NestedFilters, hasArchivedFilter, hasDeletedFilter, hasTypeFilter)
+	}
+	// build And filter based on original Or filter and default filters
+	if filters[0].Operator == model.BlockContentDataviewFilter_Or {
+		filters = addDefaultFilters(filters, hasArchivedFilter, hasDeletedFilter, hasTypeFilter)
+		return []*model.BlockContentDataviewFilter{
+			{
+				Operator:      model.BlockContentDataviewFilter_And,
+				NestedFilters: filters,
+			},
+		}
+	}
+	return filters
+}
+
+func addDefaultFilters(filters []*model.BlockContentDataviewFilter, hasArchivedFilter, hasDeletedFilter, hasTypeFilter bool) []*model.BlockContentDataviewFilter {
+	if !hasArchivedFilter {
+		filters = append(filters, &model.BlockContentDataviewFilter{RelationKey: bundle.RelationKeyIsArchived.String(), Condition: model.BlockContentDataviewFilter_NotEqual, Value: pbtypes.Bool(true)})
+	}
+	if !hasDeletedFilter {
+		filters = append(filters, &model.BlockContentDataviewFilter{RelationKey: bundle.RelationKeyIsDeleted.String(), Condition: model.BlockContentDataviewFilter_NotEqual, Value: pbtypes.Bool(true)})
+	}
+	if !hasTypeFilter {
+		// temporarily exclude Space objects from search if we don't have explicit type filter
+		filters = append(filters, &model.BlockContentDataviewFilter{RelationKey: bundle.RelationKeyType.String(), Condition: model.BlockContentDataviewFilter_NotIn, Value: pbtypes.Float64(float64(model.ObjectType_space))})
+	}
+	return filters
+}
+
+func hasDefaultFilters(filters []*model.BlockContentDataviewFilter) (bool, bool, bool) {
```

Comment:
why did you split hasDefaultFilters and addDefaultFilters?

Reply:
Both are used in `injectDefaultFilters` along with addDefaultFiltersToNested. I have separate logic `addDefaultFiltersToNested` and `addDefaultFilters`. To avoid unnecessary level in filter